policy_agent/policy_agent.py

```diff
@@ -0,0 +1,655 @@
+                                  kind=KIND_NAMESPACE,
+                                  target=ns)
+
+        for svc in initServices["items"]:
+            self.process_resource(action=CMD_ADDED,
+                                  kind=KIND_SERVICE,
+                                  target=svc)
+
+        for pod in initPods["items"]:
+            self.process_resource(action=CMD_ADDED,
+                                  kind=KIND_POD,
+                                  target=pod)
+
+        for ep in initEndpoints["items"]:
+            self.process_resource(action=CMD_ADDED,
+                                  kind=KIND_ENDPOINTS,
+                                  target=ep)
+
+        self.nsResourceVersion = initEndpoints["metadata"]["resourceVersion"]
+        self.svcResourceVersion = initServices["metadata"]["resourceVersion"]
+        self.poResourceVersion = initPods["metadata"]["resourceVersion"]
+        self.epResourceVersion = initNamespaces["metadata"]["resourceVersion"]
+
+        _log.info("Service Resource Version = %s" % self.svcResourceVersion)
+        _log.info("Namespace Resource Version = %s" % self.nsResourceVersion)
+        _log.info("Pod Resource Version = %s" % self.poResourceVersion)
+        _log.info("Endpoints Resource Version = %s" % self.epResourceVersion)
+
+    def init_threads(self):
+        # Assert these have not been initialized yet.
+        assert not self.namespace_watcher
+        assert not self.service_watcher
+        assert not self.pod_watcher
+        assert not self.endpoints_watcher
+
+        self.namespace_watcher = Thread(target=_keep_watch,
+                                        args=(self.watcher_queue, "namespaces", self.nsResourceVersion))
+        self.service_watcher = Thread(target=_keep_watch,
+                                      args=(self.watcher_queue, "services", self.svcResourceVersion))
+        self.pod_watcher = Thread(target=_keep_watch,
+                                  args=(self.watcher_queue, "pods", self.poResourceVersion))
+        self.endpoints_watcher = Thread(target=_keep_watch,
+                                        args=(self.watcher_queue, "endpoints", self.epResourceVersion))
+
+        self.namespace_watcher.daemon = True
+        self.service_watcher.daemon = True
+        self.pod_watcher.daemon = True
+        self.endpoints_watcher.daemon = True
+
+    def read_updates(self):
+        """
+        Pulls an update off the Queue and processes it.
+        If no responses remain in the queue, it will trigger a resync for all resource lists.
+        """
+        try:
+            response = self.watcher_queue.get_nowait()
+            r_json = json.loads(response)
+            r_type = r_json["type"]
+            r_obj = r_json["object"]
+            r_kind = r_obj["kind"]
+
+            _log.debug("Reading update %s: %s" % (r_type, r_kind))
+
+            if r_kind in VALID_KINDS:
+                if r_type == CMD_DELETED:
```

Comment:
Mismatch in terminology here - Is it a type, or a command?

I think this should be `cmd` instead of `r_type` - it's important to keep your naming consistent for more readable code.